CMakeLists.txt

```diff
@@ -260,6 +261,20 @@ IF(WANT_PORTAUDIO)
 	ENDIF(PORTAUDIO_FOUND)
 ENDIF(WANT_PORTAUDIO)
 
+# check for libsoundio
+IF(WANT_SOUNDIO)
+  FIND_PACKAGE(SoundIo)
+  IF(SOUNDIO_FOUND)
+    SET(LMMS_HAVE_SOUNDIO TRUE)
+    SET(STATUS_SOUNDIO "OK")
+    INCLUDE_DIRECTORIES("${SOUNDIO_INCLUDE_DIR}")
```

Comment:
> The first thing I tried was using PortAudio's CMake lines as a template but that lead to inability to build if the library is not installed. Instead of trying to figure out why that wasn't working, I opted to just do it how I do it in my own projects, since I know that will work.


In regards to the CMakeLists lines, I'll be the odd man out and say I'd prefer we keep this consistent -- speaking less about where it *should be* but rather on behalf of the [recent cleanup efforts](https://github.com/LMMS/lmms/pull/2210/files).

From my experience, variable scope and order of operations is what seems to cause most `CMakeLists.txt` problems.  At a glance, `LMMS_REQUIRED_LIBS` adds `${PORTAUDIO_LIBRARIES}` to parent scope.  In my experience, the `PARENT_SCOPE` related issues are usually the cause of `ADD_SUBDIRECTORY` headaches and this is because a local variable in cmake isn't exposed to its parent.

Anyway, happy to help. :+1: